tests/tests.rs

```diff
@@ -13,119 +13,154 @@ extern crate cargo;
 #[macro_use]
 extern crate serde_json;
 
+use std::sync::{Arc, Condvar, Mutex};
+use std::thread;
+use std::time::Duration;
+
 mod support;
 use support::{basic_bin_manifest, project};
 use support::harness::{ExpectedMessage, RlsHandle};
 
+fn timeout<F>(dur: Duration, func: F)
+    where F: FnOnce() + Send + 'static {
+    let pair = Arc::new((Mutex::new(false), Condvar::new()));
```

Comment:
Making this pair doesn't seem worthwhile to me, you could just clone the mutex and condvar separately in fewer lines of code.

Reply:
I'm not sure what you mean here.  I don't actually want to clone the mutex and condvar and I don't see how I could do it in less lines.  BTW, this code follows the rust stdlib example code for how to work with CondVar.

Reply:
I meant you could write:
```
let mutex = Arc::new(Mutex::new(...));
let condvar = Arc::new(Condvar::new());
```

(and clone the `Arc`s, not the mutex/condvar). I guess having a pair saves an `Arc` but it seems a bit convoluted to me.

Reply:
Either way is fine, though.

Reply:
Read the stdlib regarding `Condvar` + examples and additionally pthread API - this is surprisingly unergonomic to use! I tried to simplify it somehow, but couldn't. I'd expect it to be much easier to run a task in thread with a timeout, but it feels like one has to do the all the machinery by hand, just like in C.